cni/pkg/repair/repaircontroller.go

```diff
@@ -69,16 +69,27 @@ func NewRepairController(reconciler brokenPodReconciler) (*Controller, error) {
 
 	_, c.podController = cache.NewInformer(podListWatch, &v1.Pod{}, 0, cache.ResourceEventHandlerFuncs{
 		AddFunc: func(newObj interface{}) {
-			c.workQueue.AddRateLimited(newObj)
+			c.mayAddToWorkQueue(newObj)
 		},
 		UpdateFunc: func(_, newObj interface{}) {
-			c.workQueue.AddRateLimited(newObj)
+			c.mayAddToWorkQueue(newObj)
 		},
 	})
 
 	return c, nil
 }
 
+func (rc *Controller) mayAddToWorkQueue(obj interface{}) {
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		repairLog.Error("Cannot convert object to pod. Skip adding it to the repair working queue.")
+		return
+	}
+	if rc.reconciler.detectPod(*pod) {
```

Comment:
confused how this can happen  - shouldn't the fieldselector already filter these out?

Doing this type of clientside filtering will likely not have a huge impact, as we still need to process all pods in memory. FieldSelector is server side